libethereum/BlockChainSync.cpp

```diff
@@ -177,6 +177,15 @@ BlockChainSync::~BlockChainSync()
 	abortSync();
 }
 
+void BlockChainSync::onBlockImported(BlockHeader const& _info)
+{
+	//if a block has been added via mining or other block import function
+	//through RPC, then we should count it as a last imported block
+	RecursiveGuard l(x_sync);
+	m_lastImportedBlock = static_cast<unsigned>(_info.number());
```

Comment:
Also, please add `m_highestBlock = max(m_lastImportedBlock, m_highestBlock)` - now I see that this is what `onPeerNewBlock` does after successful import and here the same should happen I suppose